website/docs/plugin/framework/handling-data/dynamic-data.mdx

```diff
@@ -0,0 +1,167 @@
+---
+page_title: 'Plugin Development - Framework: Handling Data - Dynamic Data'
+description: >-
+  How to handle data when utilizing dynamic attributes or dynamic function parameters.
+---
+
+# Dynamic Data
+
+<Note>
+
+Dynamic data handling information is only relevant when utilizing a [dynamic attribute](/terraform/plugin/framework/handling-data/attributes/dynamic), [dynamic parameter](/terraform/plugin/framework/functions/parameters/dynamic) or [dynamic return](/terraform/plugin/framework/functions/returns/dynamic).
+
+Static types should always be preferred over dynamic types, when possible.
+
+</Note>
+
+Dynamic data handling uses the [framework dynamic type](/terraform/plugin/framework/handling-data/types/dynamic) to communicate to Terraform that the value type of an [attribute](/terraform/plugin/framework/handling-data/attributes/dynamic), [function parameter](/terraform/plugin/framework/functions/parameters/dynamic), or [function return](/terraform/plugin/framework/functions/returns/dynamic), will be determined at runtime. This allows a provider developer to handle multiple value types of data with a single attribute/parameter/return.
+
+Using dynamic data has a negative impact on practitioner experience when using Terraform and downstream tooling, like practitioner configuration editor integrations. Dynamics do not change how [Terraform's static type system](/terraform/language/expressions/types) behaves and all data consistency rules are applied the same as static types. Provider developers should understand all the below [considerations](#considerations) when creating a provider with a dynamic type.
+
+It's recommended that provider developers only use dynamics when using one or more static types is not a suitable alternative.
```

Comment:
Nit: This is essentially reiterating the note callout about. Here are stylistic suggestions to follow the HashiCorp writing guidance (e.g. no contractions), but maybe this sentence should be moved/removed and only in the callout.

```suggestion
Only use a dynamic type when there is not a suitable static type alternative.
```